### PR TITLE
Use the JHtml::_ function instead of calling functions directly.

### PR DESCRIPTION
**Pull request by okonomiyaki3000**
Pull Request for Issue # .
### Summary of Changes

Instead of calling JHtml static functions directly, call them with the underscore function like `JHtml::_('functionname');` so that it's possible to override them. 
### Testing Instructions

The changed files could potentially affect many pages (for example, any list page with a disabled ordering column). You should see no difference before and after applying the patch. 
### Documentation Changes Required

No but, if there is some documentation for developers, it would be a good idea for that documentation to make it clear that `JHtml` static functions should not be called directly but invoked through the `_` function.

### CODE REVIEW

**Review by mbabker — CHANGES_REQUESTED** [review]
Please revert the changes in `JHtmlTest`.  We want to explicitly only test the actual functions there, not that the loader is able to resolve to the correct function as well (we have tests for that part specifically already).

File: libraries/cms/html/html.php [hunk 1]
```diff
@@ -670,7 +670,7 @@ public static function stylesheet($file, $attribs = array(), $relative = false,
 	 *
 	 * @return  mixed  nothing if $path_only is false, null, path or array of path if specific js browser files were detected.
 	 *
-	 * @see     JHtml::stylesheet()
+	 * @see     JHtml::_('stylesheet', )
```
Comment (mbabker) [inline libraries/cms/html/html.php]: Can you revert this line too (and if there are any others in doc blocks, those as well)?

**Comment by okonomiyaki3000** [comment]
@mbabker tests have been reverted, please check.

**Review by mbabker — APPROVED** [review]

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 3973dfb517b6e49fa1d58d009d55fb3b4d62a914<br /><br />Tested on Menu Item `Category List` and hidden `Show Author`. Page display same woth or without PR-<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12546">issues.joomla.org/tracker/joomla-cms/12546</a>.</sub>

**Comment by dgt41** [comment]
@okonomiyaki3000 can you update your PR, it's out of sync, Thanks

**Comment by okonomiyaki3000** [comment]
Should be in sync now.

**Comment by ThomasFinnern** [comment]
I have tested this item :white_check_mark: successfully on ae82474492857da4bcfc465777890f27767cd3e6<br /><br />Please check whether the operations performed meet the requirements for this test (newbee on testing)

On new installation PHP7.1.1 and Joomla! 3.7.0-beta2 dev [ Amani ] 2-February-2017 18:53 GMT: 
- Added 5 contacts. Looked at the conact list. Reodered the list, changed the status and reordered it by status
- Applied the patch and did the same steps. No visal changes seen.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12546">issues.joomla.org/tracker/joomla-cms/12546</a>.</sub>

**Comment by zero-24** [comment]
I have just fixed the conflicts. -> RTC

**Pull request merged**